setup.cfg

```
@@ -41,6 +41,7 @@ install_requires =
     jsonschema>=3.0.0  # for utils
     jsonpatch>=1.15
     pyyaml>=5.1  # for parsing CLI equal-delimited options
+    importlib_resources; python_version < "3.9"
```

It is always good to add a lower bound for dependencies to make it explicit and easy for `pip` to do a correct solve. As we need `importlib_resources.as_file` and `importlib_resources.files` (and I think that's it) here then we should add the lowest bound for that functioning API. From looking at the [`importlib_resources` docs](https://importlib-resources.readthedocs.io/en/latest/history.html#v1-3-0) and testing this out myself the first time the `importlib_resources.as_file` exists is `v1.3.0`.

```suggestion
    importlib_resources>=1.3.0; python_version < "3.9"  # for resources in schema
```

We should then additionally add

```
importlib_resources==1.3.0
```

to `tests/constraints.txt` just after `pyyaml`

https://github.com/scikit-hep/pyhf/blob/1884c6c08bc5b629724dd9ef3c60ec4386353b94/tests/constraints.txt#L1-L7